universal-application-tool-0.0.1/app/controllers/admin/ProgramAdminManagementController.java

```diff
@@ -86,10 +87,29 @@ public Result update(Http.Request request, long programId) {
           roleService.makeProgramAdmins(
               programId, ImmutableSet.copyOf(manageAdminForm.getAdminEmails()));
       Result result = redirect(routes.AdminProgramController.index());
-      if (maybeError.isPresent()) {
-        return result.flashing("error", maybeError.get().message());
+
+      if (!maybeError.isPresent()){
+        return result;
       }
-      return result;
+
+      Optional<Program> program =
+          programRepository.lookupProgram(programId).toCompletableFuture().join();
+
+      if (program.isEmpty()) {
+        return notFound(String.format("Program with ID %s was not found", programId));
+      } else {
+        ImmutableList<String> programAdmins =
+            programRepository.getProgramAdministrators(programId).stream()
+                .map(Account::getEmailAddress)
+                .collect(toImmutableList());
+
+        return ok(
+            manageAdminsView.render(
+                request,
+                program.get().getProgramDefinition(),
+                programAdmins,
+                Optional.of(maybeError.get().message())));
+      }     
```

Comment:
suggestion (non-blocking): could maybe factor this out to reduce duplicated code with `edit(`

Reply:
Good idea, done!